libraries/AC_Avoidance/AP_Astar_RT.h

```diff
@@ -0,0 +1,150 @@
+#pragma once
+
+#include <AP_Common/AP_Common.h>
+#include <AP_Common/Location.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_HAL/AP_HAL.h>
+#include "AP_OAVisGraph.h"
+#include <AC_Avoidance/AP_OADatabase.h>
+
+#define OA_RTASTAR_NUM_POINTS_PER_OBJECT 8
+#define OA_RTASTAR_DEBUG_ENABLE 0
+
+/*
+ * Real Time A* avoidance algorithm for avoiding the polygon and circular fence and dynamic objects detected by the proximity sensor
+ */
+class AP_OART_AStar {
+public:
+    AP_OART_AStar();
+
+    /* Do not allow copies */
+    AP_OART_AStar(const AP_OART_AStar &other) = delete;
+    AP_OART_AStar &operator=(const AP_OART_AStar&) = delete;
+
+    // send configuration info stored in front end parameters
+    void set_config(float margin_max) { _margin_max_cm = MAX(margin_max, 0.0f)*100.0f;  }
```

Comment:
this is the equivalent of Dijkstra's, "set_fence_margin" method so it would be good to keep the two names consistent.  Either name is fine but if we go with a new name, let's rename Dijkstra's to match as well.

Reply:
Changed Dijkstras to match with BendyRuler as well